### DISCUSSION TITLE
啥时热更新可以支持新增方法、删除方法、新增字段、删除字段也 OK？

### DISCUSSION THREAD

**Discussion by vampire007** [alibaba/arthas#2254] [Ideas]
redefine 和 retransform 命令啥时热更新可以支持新增方法、删除方法、新增字段、删除字段也被允许呢？通常情况下，这是有极大的诉求的，实际场景中，不可能不对类进行大刀阔斧的修改。~

**Comment by hengyunabc** [comment]
没有办法，jvm自身的限制。
